engine/src/main/java/io/camunda/zeebe/engine/api/RecordProcessor.java

```diff
@@ -63,7 +63,22 @@
   ProcessingResult process(TypedRecord record, ProcessingResultBuilder processingResultBuilder);
 
   /**
-   * Called by platform when a processing error occurred
+   * Called by platform when a processing error occurred.
+   *
+   * <p><em>Contract</em>
+   *
+   * <ul>
+   *   <li>Record will be a command
+   *   <li>Will be called if an uncaught exception is thrown in {@link #process(TypedRecord,
```

Comment:
:wrench: : Will also be called when an uncaught exception is thrown in ProcessingStateMachine during commit of the transaction